### DISCUSSION TITLE
Skywalking version: 9.7.0, added backend Spring boot version: 2.7.12, error reported

### DISCUSSION THREAD

**Discussion by cfy3133** [apache/skywalking#12351] [General]
After Java joined Skywalking, the agent reported an error:main SkyWalkingAgent : Enhance class org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor error，Then the connected application reported an error:[main] [empty#0 on NoUser] ERROR org.springframework.boot.SpringApplication:821 - Application run failed org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.context.annotation.internalConfigurationAnnotationProcessor': Instantiation of bean failed; nested exception is java.lang.ClassFormatError: Duplicate field name "cachedValue 𝑠 𝑤 swho7p9v2$7stfck1" with signature "Ljava.lang.reflect.Method;" in class file org/springframework/beans/BeanWrapperImpl。

The possible issue is that when the SkyWalking Agent attempts to enhance the BeanWrapperImpl class, the additional field names it generates overlap with the existing field names within the class, resulting in ClassFormatError. This is usually caused by the incompatibility between SkyWalking's dynamic bytecode injection mechanism and certain internal details of the Spring framework.

**Comment by wu-sheng** [comment]
It doesn't seem so. More likely you have spring devtool or similar things to duplicate change codes, to trigger skywalking kernel conflicts.

**Comment by cfy3133** [comment]
Okay, thank you for the author's reply. May I ask if removing the hot deployment in devtools can solve the problem

**Reply by wu-sheng** [reply]
About devtool, it is.
But I can't tell whether there are more things to effect.

**Reply by cfy3133** [reply]
> 关于devtool，它是。但我无法判断是否还有更多事情要影响。
Okay, thank you for the author's patient reply. I'll give it a try

**Comment by cfy3133** [comment]
Still the same error, probably due to what the author said and other similar things like copying and changing code

**Comment by cfy3133** [comment]
![image](https://github.com/apache/skywalking/assets/43978367/810cc2ce-8836-42fa-8e78-e7cbf7f8f731)

**Reply by wu-sheng** [reply]
Someone fixed the compatibility issue with Arthas in theory, but we can't guarantee that. Arthas is maintained by Alibaba team, we can't control how it changes.

You only mentioned `Skywalking version: 9.7.0`, but this is actually not an agent version. So, you could try the latest agent. But I am not sure whether they are good or not.

**Reply by cfy3133** [reply]
Okay, thank you
